cli/azd/pkg/apphost/generate.go

```diff
@@ -274,9 +275,50 @@ func newInfraGenerator() *infraGenerator {
 	}
 }
 
+// pullParentSettings check a special case where:
+// - resource has a parent with type `container.v0`
+// When this is true, the resource should inherit the parent settings (type, bindings, env, inputs, etc.)
+// For example, a `postgres.database.v0` resource that has a parent of type `container.v0` should inherit the
+// container settings from the parent and become a `container.v0` resource.
+func pullParentSettings(resourceName string, resource *Resource, manifest *Manifest) *Resource {
```

Comment:
Could you write a few tests for this method?  Either table driven or direct - would just be helpful to see how things flow and the changes to the env and connection string.